src/test/java/uk/gov/justice/hmpps/prison/aop/connectionproxy/OracleConnectionAspectTest.kt

```diff
@@ -218,15 +196,17 @@ class OracleConnectionAspectTest {
       fun `should set Nomis context and schema`() {
         connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection)
 
-        verify(pooledConnection).prepareStatement(
-          check {
-            assertThat(it).contains("nomis_context.set_context('AUDIT_MODULE_NAME', 'MERGE')")
-            assertThat(it).doesNotContain("nomis_context.set_context('AUDIT_USER_ID'")
-            assertThat(it).doesNotContain("nomis_context.set_client_nomis_context(")
-          }
-        )
-        verify(pooledConnection).prepareStatement("ALTER SESSION SET CURRENT_SCHEMA=some default schema")
-        verify(pooledPreparedStatement, times(2)).execute()
+        verify(nomisConfigurer).setSuppressXtagEvents(any<ResettableContextConnection>())
+        verify(nomisConfigurer).setDefaultSchema(any<ResettableContextConnection>())
+      }
+
+      @Test
+      fun `should open resettable connection if suppress Xtags`() {
+        whenever(MDC.get(MdcUtility.SUPPRESS_XTAG_EVENTS)).thenReturn("true")
+
+        connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection)
+
+        verify(nomisConfigurer).setDefaultSchema(any<ResettableContextConnection>())
```

Comment:
This is a significant change - we now use a resettable connection when suppressing XTag events.